truffle.js

```diff
@@ -56,7 +56,17 @@ const networkCandidates = {
             if (mnemonic != null) {
                 return new HDWalletProvider(mnemonic, url);
             }
-            return new HDWalletProviderPrivkey(privateKey.toLowerCase(), url);
+            const privateKeyArray = privateKeys
+                .replace(/^\s+|\s+$/g, "").split(/\s+/g)
+                .map(k => {
+                    if (!k.match(/^[0-9a-f]+$/i)) {
```

Comment:
Is address starts with `0x` possible to come in here?

Reply:
As `k` is not an address but a private key, the prefix `0x` is not prepended to it.